redis_hashring/__init__.py

```diff
@@ -315,13 +324,11 @@ def poll(self):
 
         try:
             while True:
-                # Since Redis' listen method blocks, we use select to inspect the
-                # underlying socket to see if there is activity.
-                fileno = pubsub.connection._sock.fileno()
                 timeout = max(0, POLL_INTERVAL - (time.time() - last_heartbeat))
-                r, w, x = self._select([fileno], [], [], timeout)
-                if fileno in r:
-                    next(gen)
+                message = pubsub.get_message(
```

Comment:
Would it be an improvement if we turned this into a loop to avoid unnecessary updates?

```
has_messages = False
while pubsub.get_message(ignore_subscribe_messages=True, timeout=timeout):
    has_messages = True
if has_messages:
    self.update()
```

This can be implemented separately since I believe your change is equivalent, so we should merge this PR first before improving it.

Reply:
I think I like @jkemp101's implementation better, from https://github.com/closeio/redis-hashring/pull/5. I added it to this PR, please review to see if it accomplishes what you intended.